src/lib/collision_prevention/CollisionPrevention.cpp

```diff
@@ -197,80 +351,60 @@ CollisionPrevention::_enterData(int map_index, float sensor_range, float sensor_
 	return false;
 }
 
-void
-CollisionPrevention::_updateObstacleMap()
+bool
+CollisionPrevention::_checkSetpointDirectionFeasability()
 {
-	_sub_vehicle_attitude.update();
+	bool setpoint_feasible = true;
 
-	// add distance sensor data
-	for (auto &dist_sens_sub : _distance_sensor_subs) {
-		distance_sensor_s distance_sensor;
+	for (int i = 0; i < BIN_COUNT; i++) {
+		// check if our setpoint is either pointing in a direction where data exists, or if not, wether we are allowed to go where there is no data
+		if ((_obstacle_map_body_frame.distances[i] == UINT16_MAX && i == _setpoint_index) && (!_param_cp_go_nodata.get()
+				|| (_param_cp_go_nodata.get() && _data_fov[i]))) {
+			setpoint_feasible =  false;
 
-		if (dist_sens_sub.update(&distance_sensor)) {
-			// consider only instances with valid data and orientations useful for collision prevention
-			if ((getElapsedTime(&distance_sensor.timestamp) < RANGE_STREAM_TIMEOUT_US) &&
-			    (distance_sensor.orientation != distance_sensor_s::ROTATION_DOWNWARD_FACING) &&
-			    (distance_sensor.orientation != distance_sensor_s::ROTATION_UPWARD_FACING)) {
-
-				// update message description
-				_obstacle_map_body_frame.timestamp = math::max(_obstacle_map_body_frame.timestamp, distance_sensor.timestamp);
-				_obstacle_map_body_frame.max_distance = math::max(_obstacle_map_body_frame.max_distance,
-									(uint16_t)(distance_sensor.max_distance * 100.0f));
-				_obstacle_map_body_frame.min_distance = math::min(_obstacle_map_body_frame.min_distance,
-									(uint16_t)(distance_sensor.min_distance * 100.0f));
-
-				_addDistanceSensorData(distance_sensor, Quatf(_sub_vehicle_attitude.get().q));
-			}
 		}
 	}
 
-	// add obstacle distance data
-	if (_sub_obstacle_distance.update()) {
-		const obstacle_distance_s &obstacle_distance = _sub_obstacle_distance.get();
-
-		// Update map with obstacle data if the data is not stale
-		if (getElapsedTime(&obstacle_distance.timestamp) < RANGE_STREAM_TIMEOUT_US && obstacle_distance.increment > 0.f) {
-			//update message description
-			_obstacle_map_body_frame.timestamp = math::max(_obstacle_map_body_frame.timestamp, obstacle_distance.timestamp);
-			_obstacle_map_body_frame.max_distance = math::max(_obstacle_map_body_frame.max_distance,
-								obstacle_distance.max_distance);
-			_obstacle_map_body_frame.min_distance = math::min(_obstacle_map_body_frame.min_distance,
-								obstacle_distance.min_distance);
-			_addObstacleSensorData(obstacle_distance, Quatf(_sub_vehicle_attitude.get().q));
-		}
-	}
+	return setpoint_feasible;
+}
 
-	// publish fused obtacle distance message with data from offboard obstacle_distance and distance sensor
-	_obstacle_distance_pub.publish(_obstacle_map_body_frame);
+void
+CollisionPrevention::_transformSetpoint(const Vector2f &setpoint)
+{
+	_setpoint_dir = setpoint / setpoint.norm();;
+	const float sp_angle_body_frame = atan2f(_setpoint_dir(1), _setpoint_dir(0)) - _vehicle_yaw;
```

Comment:
```suggestion
	const float sp_angle_body_frame = atan2f(setpoint(1), setpoint(0)) - _vehicle_yaw;
```
The length of the vector doesn't matter to determine its direction.

Reply:
c78f29331b7a99fe95d20af30b772006aa78cb29